### PR TITLE
Update RELEASE.txt

### LINKED ISSUE #5423: CI has been updated to build on tags; RELEASE.txt needs to be updated
See https://github.com/scikit-image/scikit-image/pull/5397

### PR DESCRIPTION
**Pull request by leej3**
Resolves #5423

This is the release workflow as I now understand it to be.

Feedback on the implementation itself also welcome (now that it is more
explicitly documented).

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: RELEASE.txt [hunk 1]
```diff
@@ -71,122 +68,57 @@ access), specifically to:
...
 
-    1. Back on the release branch, update the version number to stable in
-       ``skimage/__init__.py``, commit, and push.
+    git tag -m <github_release_message> buildwheel_0.19.0test
 
-    2. Add the version number as a tag in git::
+  2. Push the new tag to GitHub (you can delete these after if you wish)::
 
-        git tag -s -m <github_release_message> [-u <key-id>] v<major>.<minor>.0
+    git push upstream buildwheel_0.19.0test
```
Comment (mkcor) [inline RELEASE.txt:82]: Shouldn't this be something like
```suggestion
    git push upstream buildwheel_<major>.<minor>.0test
```
?

Comment (leej3) [reply RELEASE.txt]: yes that's better. thanks

File: RELEASE.txt [hunk 2]
```diff
@@ -71,122 +68,57 @@ access), specifically to:
...
+  and testing of the wheels by CI is to your satisfaction.
 
 
-- Perform the following step first on a release candidate to check the wheels
-  are building fine. Then proceed with the release.
+  1. Add a tag in git that starts with the string "buildwheel"::
 
-    1. Back on the release branch, update the version number to stable in
-       ``skimage/__init__.py``, commit, and push.
+    git tag -m <github_release_message> buildwheel_0.19.0test
```
Comment (grlee77) [inline RELEASE.txt:78]:
```suggestion
    git tag -m <github_release_message> buildwheel_<major>.<minor>.0test
```

use major/minor as in step 2 below

**Review by grlee77 — CHANGES_REQUESTED** [review]
I requested one small change, but this otherwise looks good to me

**Review by grlee77 — APPROVED** [review]

**Pull request merged**